src/lang/base/Utils.ml

```diff
@@ -64,7 +68,11 @@ module AssocDictionary : Dictionary = struct
   let rec update k v d =
     match d with
     | []               -> []
-    | (kd, vd) :: rest -> if k = kd then (k, v) :: rest else (kd, vd) :: (update k v rest)
+    | (kd, vd) :: rest -> if k = kd then (k, v) :: (update k v rest) else (kd, vd) :: (update k v rest)
```

Comment:
This change will also break the cashflow analyser in the same situation as above.

Reply:
Can we instead add two new functions `remove_all` and `update_all`?